lighthouse-core/gather/gatherers/css-usage.js

```diff
@@ -11,38 +11,79 @@ const FRGatherer = require('../../fraggle-rock/gather/base-gatherer.js');
  * @fileoverview Tracks unused CSS rules.
  */
 class CSSUsage extends FRGatherer {
+  constructor() {
+    super();
+    /** @type {Array<LH.Crdp.CSS.StyleSheetAddedEvent>} */
+    this._stylesheets = [];
+    /** @param {LH.Crdp.CSS.StyleSheetAddedEvent} sheet */
+    this._onStylesheetAdded = sheet => this._stylesheets.push(sheet);
+    /** @type {LH.Crdp.CSS.RuleUsage[]|undefined} */
+    this._ruleUsage = undefined;
```

Comment:
This should never always be set by the time we use it, which is why I opted to throw an error on `undefined` rather than initialize to `[]`. I'm open to changing it though.